pyproject.toml

```diff
@@ -0,0 +1,9 @@
+[build-system]
+requires = [
+    "wheel",
+    "setuptools",
+    "Cython>=0.23",
+    "numpy==1.10.4; python_version<='3.5'",
+    "numpy==1.12.1; python_version=='3.6'",
+    "numpy==1.13.1; python_version>='3.7'",
```

Comment:
Thinking about this in the context of wheel builds being made available, should we be bumping the `numpy` requirement at run time based on what version we used at build time or does this somehow happen for us?

Reply:
bleh! Wheels are a pain!

Reply:
I do not think this happens automatically, but I'm not sure if it is necessary in practice.  I have asked for some advice over at https://github.com/PyWavelets/pywt/pull/321#issuecomment-377542922

I [copied these particular values from scipy](https://github.com/pypa/pip/issues/4582#issuecomment-312111015), but bumped 1.9.3 to 1.10.4 to reflect our minimum NumPy requirement.  From what I can tell, SciPy still uses 1.8.2 in their `install_requires` in setup.py, but I'm not sure if there may be some other magic being performed elsewhere.

Reply:
The concern would be if a wheel was built with a newer NumPy, but allowed to be installed on an older one where expected symbols are missing.

Looking at the pip thread linked in your pywt comment, I think they are relying on the fact that each version of Python comes online with a specific NumPy version to effectively impose this requirement. IOW NumPy 1.12 is the first version that supports Python 3.6. So if we build against NumPy 1.12 with Python 3.6, no older version of NumPy could be installed on Python 3.6. If I'm reading it correctly, then I think what you added here is correct. Happy to have someone else more involved with wheels confirm this though.

Yeah, that matches what the SciPy wheels show as well.

Reply:
> IOW NumPy 1.12 is the first version that supports Python 3.6. So if we build against NumPy 1.12 with Python 3.6, no older version of NumPy could be installed on Python 3.6.

That's correct, and this PR looks safe to me. However to make it more explicit, one could add an `install_requires={'numpy>=build_version'}`, where build version then needs to be determined at `setup.py` execution time based on the Python version.